File: copilot_minimax_core/patching.py
"""Built-in extension.js patch logic."""

from __future__ import annotations

from .constants import ANTHROPIC_URL, MINIMAX_MODELS, MINIMAX_URL, OLD_GET_ALL_MODELS_PREFIX
from .model_catalog import new_get_all_models_function


def patch_content(content: str) -> tuple[str, list[str]]:
    """Apply all built-in patches and return (patched_content, changes)."""
    changes: list[str] = []

    if ANTHROPIC_URL in content:
        content = content.replace(ANTHROPIC_URL, MINIMAX_URL)
        changes.append(f"Redirected {ANTHROPIC_URL} -> {MINIMAX_URL}")

    bare_old = '"api.anthropic.com"'
    bare_new = '"api.minimax.io/anthropic"'
    if bare_old in content:
        content = content.replace(bare_old, bare_new)
        changes.append("Fixed bare hostname in telemetry attributes")

    if OLD_GET_ALL_MODELS_PREFIX in content:
        content = content.replace(
            OLD_GET_ALL_MODELS_PREFIX,
            new_get_all_models_function(),
            1,
        )
        changes.append(
            "Replaced model listing with hardcoded: " + ", ".join(MINIMAX_MODELS.keys())
        )

    return content, changes


def is_patched(content: str) -> bool:
    """Check if extension.js already contains key MiniMax patch markers."""
    return MINIMAX_URL in content and "MiniMax-M2.7" in content

File: packages/copilot-minimax-cli/src/patching.js
import {
  ANTHROPIC_URL,
  MINIMAX_MODELS,
  MINIMAX_URL,
  OLD_GET_ALL_MODELS_PREFIX,
} from './constants.js';
import { newGetAllModelsFunction } from './modelCatalog.js';

export function patchContent(content) {
  const changes = [];
  let patched = content;

  if (patched.includes(ANTHROPIC_URL)) {
    patched = patched.replaceAll(ANTHROPIC_URL, MINIMAX_URL);
    changes.push(`Redirected ${ANTHROPIC_URL} -> ${MINIMAX_URL}`);
  }

  if (patched.includes('"api.anthropic.com"')) {
    patched = patched.replaceAll('"api.anthropic.com"', '"api.minimax.io/anthropic"');
    changes.push('Fixed bare hostname in telemetry attributes');
  }

  if (patched.includes(OLD_GET_ALL_MODELS_PREFIX)) {
    patched = patched.replace(OLD_GET_ALL_MODELS_PREFIX, newGetAllModelsFunction());
    changes.push(`Replaced model listing with hardcoded: ${Object.keys(MINIMAX_MODELS).join(', ')}`);
  }

  return { content: patched, changes };
}

export function isPatched(content) {
  return content.includes(MINIMAX_URL) && !content.includes(ANTHROPIC_URL);
}
